components/datetime/src/neo.rs

```diff
@@ -555,22 +555,66 @@ where
     fn try_new_internal<P, L>(
         provider: &P,
         loader: &L,
-        prefs: DateTimeFormatterPreferences,
+        mut prefs: DateTimeFormatterPreferences,
         field_set: CompositeFieldSet,
     ) -> Result<Self, DateTimeFormatterLoadError>
     where
         P: ?Sized + AllAnyCalendarFormattingDataMarkers<FSet>,
         L: DecimalFormatterLoader + AnyCalendarLoader,
     {
-        let calendar = AnyCalendarLoader::load(loader, (&prefs).into())
-            .map_err(DateTimeFormatterLoadError::Data)?;
+        match prefs.calendar_algorithm {
+            Some(
+                CalendarAlgorithm::Buddhist
+                | CalendarAlgorithm::Chinese
+                | CalendarAlgorithm::Coptic
+                | CalendarAlgorithm::Dangi
+                | CalendarAlgorithm::Ethioaa
+                | CalendarAlgorithm::Ethiopic
+                | CalendarAlgorithm::Gregory
+                | CalendarAlgorithm::Hebrew
+                | CalendarAlgorithm::Hijri(_)
+                | CalendarAlgorithm::Indian
+                | CalendarAlgorithm::Japanese
+                | CalendarAlgorithm::Persian
+                | CalendarAlgorithm::Roc,
+            ) => {}
+            // Calendar not supported by DateTimeFormatter
+            // Currently this is CalendarAlgorithm::Iso8601
+            // Let AnyCalendarKind constructor select an appropriate fallback
+            _ => prefs.calendar_algorithm = None,
+        }
+        let kind = AnyCalendarKind::new((&prefs).into());
+        // AnyCalendarKind constructor should not select an unsupported fallback
```

Comment:
this is unfortunate

Reply:
Yeah, a way to fix this would be for `icu_calendar` to export an enum `LocaleDefaultCalendarKind` which we can exhaustively match over here. But this is fine. I do prefer for this to be a `debug_assert` instead of an `assert`.